ProjFS.Mac/PrjFSKextTests/FileSystemMountPointTests.mm

```diff
@@ -0,0 +1,106 @@
+#include "../PrjFSKext/FileSystemMountPoints.hpp"
+#include "../PrjFSKext/FileSystemMountPointsPrivate.hpp"
+#import <XCTest/XCTest.h>
+
+// dummy implementation of the opaque struct
+struct mount
+{};
+
+static mount s_dummyMounts[MaxUsedMountPoints];
+
+static void FillAllMountPointSlots()
+{
+    for (unsigned i = 0; i < MaxUsedMountPoints; ++i)
+    {
+        s_usedMountPoints[i].authCacheDisableCount = 1;
+        s_usedMountPoints[i].mountPoint = &s_dummyMounts[i];
+    }
+}
+
+static void FillAllMountPointSlotsExcept(unsigned leaveEmptyIndex)
+{
+    for (unsigned i = 0; i < MaxUsedMountPoints; ++i)
+    {
+        if (i != leaveEmptyIndex)
+        {
+            s_usedMountPoints[i].authCacheDisableCount = 1;
+            s_usedMountPoints[i].mountPoint = &s_dummyMounts[i];
+        }
+    }
+}
+
+
+@interface FileSystemMountPointTests : XCTestCase
+
+@end
+
+@implementation FileSystemMountPointTests
+
+- (void)setUp
+{
+    memset(s_usedMountPoints, 0, sizeof(s_usedMountPoints));
+}
+
+- (void)tearDown
+{
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+}
+
+- (void)testFindEmptyMountPointSlot_AllFull
```

Comment:
I've added a few unit tests for the new code with this latest push. Coverage is currently not good, but I wanted to get something off the ground at least.

* For testing `MountPoint_Init()`/`MountPoint_Cleanup()` we need to implement testing versions of the `Mutex_` functions which I haven't got around to yet.
 * `MountPoint_DisableAuthCache` and `MountPoint_RestoreAuthCache` call KPIs, so I'll wait until we know what we're doing regarding mocking KPIs.